pkg/app/instances/docker.go

```diff
@@ -297,3 +310,26 @@ func (m *DockerInstanceManager) getContainerLabel(host string, key string) (stri
 	}
 	return value, nil
 }
+
+func (m *DockerInstanceManager) downloadDockerImageIfNeeded(ctx context.Context) error {
+	listRes, err := m.Client.ImageList(ctx, image.ListOptions{})
+	if err != nil {
+		return fmt.Errorf("Failed to list docker images: %w", err)
+	}
+	for _, image := range listRes {
+		for _, tag := range image.RepoTags {
+			if tag == m.Config.Docker.DockerImageName {
+				return nil
+			}
+		}
+	}
+
+	out, err := m.Client.ImagePull(ctx, m.Config.Docker.DockerImageName, image.PullOptions{})
+	if err != nil {
+		return fmt.Errorf("Failed to pull docker image '%s': %w", m.Config.Docker.DockerImageName, err)
+	}
+	defer out.Close()
```

Comment:
why do we need `out` and copying? can we just ignore the first return value of `ImagePull`?

Reply:
I thought so, but `ImagePull` API doesn't work synchronously. Without handling `out`, we cannot guarantee whether the image is pulled or not.

https://docs.docker.com/reference/api/engine/sdk/examples/

Reply:
done.